addons/website_slides/static/src/js/slides_upload.js

```diff
@@ -599,9 +659,29 @@ var SlideUploadDialog = Dialog.extend({
         var $elem = this.$(ev.currentTarget);
         var slideType = $elem.data('slideType');
         this.set('state', slideType);
-
+        if (!this.canWriteSurvey) {
+            $("#certification_give_badge").attr('disabled', true);
+        }  
         this._bindSelect2Dropdown();  // rebind select2 at each modal body rendering
     },
+
+    _canWriteSurvey: function (createUid) {
+        if (this.surveyGroup === 'manager') {
+            this.canWriteSurvey = true;
+        } else if (createUid === undefined || (this.surveyGroup === 'user' && createUid[0] === session.user_id)) {
+            this.canWriteSurvey = true;
```

Comment:
make one if with all of that
`if (cond1 || cond2 || (cond3))`
but wait ... what ? if not logged in, you can write ?

anyway, not sure that it's the proper way to check access right anyway.
I think you should check access rights in controller before calling template, give the result to the template and hide or show stuffs in the template directly.. 
Maybe I didn't understood everything here because it's vendredi soir but I think this whole thing should be re-thought.

Reply:
What do you mean, if not logged can write, you can write? :p 
creatuid is the ID of the user that created the survey and only he can modify his survey according to survey_security.
If creatuid is undefined the user is creating a new one and can write, otherwise he need to be the creator to be able to write.
I don't think I can do all this in the controller before rendering the template as at that time we don't know yet if the user wants to reuse/create a new certificate. Thus if he will be allowed to modify the current survey by adding a badge or not.

Reply:
Ah ok ! yes indeed.

Here, I would return True or False and not set a variable. So either the caller set himself the variable, either he uses directly the result of the _canWriteSurvey.